Makefile

```diff
@@ -82,57 +89,118 @@ for-all-target: $(ALL_MODS)
 integration-vet:
 	@set -e; cd tests && go vet -tags integration,testutilsintegration,zeroconfig,testutils ./... && $(GOTEST_SERIAL) $(BUILD_INFO_TESTS) -tags testutils,testutilsintegration -v -timeout 5m -count 1 ./...
 
+.PHONY: integration-test-target
+integration-test-target:
+	@set -e; cd tests && $(GOTEST_SERIAL) $(BUILD_INFO_TESTS) --tags=$(TARGET) -v -timeout 5m -count 1 ./...
+
+.PHONY: integration-test-cover-target
+integration-test-cover-target:
+	@set -e; $(MAKE_TEST_COVER_DIR) && cd tests && $(GOTEST_SERIAL) $(BUILD_INFO_TESTS) --tags=$(TARGET) -v -timeout 5m -count 1 ./... $(COVER_TESTING_INTEGRATION_OPTS)
+	$(GOCMD) tool covdata textfmt -i=$(TEST_COVER_DIR) -o ./$(TARGET)-coverage.txt
+
 .PHONY: integration-test
 integration-test:
-	@set -e; cd tests && $(GOTEST_SERIAL) $(BUILD_INFO_TESTS) --tags=integration -v -timeout 5m -count 1 ./...
+	@make integration-test-target TARGET='integration'
```

Comment:
Note: The original integration test targets are not meant to be changed in this PR. I'm simply moving them to a central target that takes the tag as the passed in `TARGET` variable, since they're all using the exact same logic otherwise.